src/lib/ext-activitylog.js

```diff
@@ -0,0 +1,160 @@
+import { save, load } from '../lib/save-load.js';
+
+export default class ActivityLog {
+  constructor() {
+    this.loadLogBtn = document.getElementById('loadLogBtn');
+    this.logInputFile = document.getElementById('logInputFile');
+    this.saveLogBtn = document.getElementById('saveLogBtn');
+    this.showLogDetailWrapper = document.getElementsByClassName(
+      'show-log-detail-wrapper'
+    )[0];
+    this.showLogDetails = document.getElementById('showLogDetails');
+    this.showLoadedLog = document.getElementById('showLoadedLog');
+    this.table = document.getElementsByTagName('table')[0];
+    this.tableFirstRow = document.querySelector('tbody tr:first-child');
+    this.notice = document.getElementsByClassName('notice')[0];
+  }
+
+  logs = [];
+  loadedLogs = [];
+
+  async getExistingLogs() {
+    const { existingLogs } = await browser.runtime.sendMessage({
+      requestType: 'sendAllLogs',
+      requestTo: 'ext-monitor',
+    });
+    return existingLogs;
+  }
+
+  async loadFile(file) {
+    try {
+      const loadedLogsAsText = await load.loadLogAsText(file);
+      const loadedJSONLogs = JSON.parse(loadedLogsAsText);
+      this.loadedLogs.push(...loadedJSONLogs);
+      this.showLoadedLog.textContent = loadedLogsAsText;
```

Comment:
The part that loads the logs from file seems to be a bit "out of place" at the moment, it doesn't really do much in this pull request and what it does isn't really that useful yet (e.g. the logs loaded are not rendered as rows in the table)

We may instead defer "load from file" to a separate pull request and think a bit more "what load from file should be really doing") and have a better and more clear plan about it, e.g.:
how is the activity log page going to serve this feature to the user?
how is the user going to switch between looking to logs loaded from file and logs being collected in that session from the monitored extensions? how is the activity log page going to may make it clear to the user what logs is looking to (loaded from file or collected in realtime)? and other similar design questions.

Reply:
Yeah, I have comment out a TODO here, which I will take care of in the next pr. :-) In the meantime, I will make a clear design answering the above questions.